src/databricks/labs/ucx/hive_metastore/views_migrator.py

```diff
@@ -0,0 +1,105 @@
+import sqlglot
+from sqlglot import ParseError
+from sqlglot.expressions import Expression as SqlExpression
+from sqlglot.expressions import Table as SqlTable
+
+from databricks.labs.ucx.hive_metastore import TablesCrawler
+from databricks.labs.ucx.hive_metastore.tables import Table
+
+
+class ViewToMigrate:
+
+    table: Table
+    table_dependencies: set[Table]
+    view_dependencies: set[Table]
+
+    def __init__(self, table: Table):
+        if table.view_text is None:
+            raise RuntimeError("Should never get there! A view must have 'view_text'!")
+        self.table = table
+        self.table_dependencies = set()
+        self.view_dependencies = set()
+
+    def compute_dependencies(self, all_tables: dict[str, Table]):
```

Comment:
i'd prefer if this method is hidden and the only method we expose is to iterate over dependencies, while we lazily call this behind the cover.

Reply:
done